### PR TITLE
Staging

### PR DESCRIPTION
**Pull request by vishwajeetgslab**
Hi @JosephPJordan updated the docs for v2.9 Release. Can you please review and merge it. Let me know anything needs to changed. Thanks

### CODE REVIEW

File: docs/l2-l3.rst [hunk 1]
```diff
@@ -92,7 +92,7 @@ Create Default Gateway
 
 4. Click :guilabel:`Submit`.
 
-The default gateway is created on the BIG-IP device and is added to the application database. For an HA setup, the default gateway will automatically sync to the BIG-IP peer device.
+The default gateway is created on the BIG-IP device and is added to the application database. For an HA setup, the default gateway will automatically sync to the BIG-IP peer device. F5 ACI ServiceCenter has added support for brownfield default gateway from v2.9 where default gateway will automatically sync to FASC database once user navigates to L2-L3 tab. FASC has added support to allow any Default Gateway name to be synced to FASC database (Until version 2.8, the only default gateway name supported for  was **apic-default-gateway**).
```
Comment (JosephPJordan) [inline docs/l2-l3.rst:95]: F5 ACI ServiceCenter v2.9 added support for a brownfield default gateway, where the default gateway automatically syncs to the FASC database once the user navigates to the L2-L3 tab. FASC added support to allow any Default Gateway name to be synced to the FASC database (in previous versions, the only supported default gateway name was **apic-default-gateway**).

File: docs/navigate.rst [hunk 2]
```diff
@@ -10,11 +10,11 @@ View VLAN table
 
    In the :guilabel:`Table` list, the VLAN table is selected by default.
 
-2. From the :guilabel:`Partition` list, select the partition you’re interested in.
+2. From the :guilabel:`Partition` list, select the partition you’re interested in. **All partition** partition is supported from v2.9 onwards for VLAN table.
 
    The table shows all the VLANs (vlan encaps) from the BIG-IP device that have a corresponding **Logical Device|Tenant** entry on the APIC.
    
-   The table shows all the VLANs (vlan encaps) from the BIG-IP device that have a corresponding **Endpoint Group|Application Profile|Tenant** entry on the APIC. These correspond to the **Static Port VLANs** and **Static Leaf VLANs** under the aforementioned **Endpoint Group**.
+   The table shows all the VLANs (vlan encaps), Self IP, Interfaces from the BIG-IP device that have a corresponding **Endpoint Group|Application Profile|Tenant** entry on the APIC. These correspond to the **Static Port VLANs** and **Static Leaf VLANs** under the aforementioned **Endpoint Group**. Users are able to view information related to Self IPs and interfaces by clicking on the dropdown icon in Self IP and interface columns respectively.
```
Comment (JosephPJordan) [inline docs/navigate.rst:17]: ..by clicking the dropdown icon in the Self IP and Interface columns respectively.

File: docs/navigate.rst [hunk 3]
```diff
@@ -24,7 +24,7 @@ View VIP table
 
 1. Click the Visibility tab, and then from the :guilabel:`Table` list, click the :guilabel:`VIP` table.
 
-2. From the :guilabel:`Partition` list, click the partition you're interested in.
+2. From the :guilabel:`Partition` list, click the partition you're interested in. **All partition** partition is supported from v2.9 onwards for VIP table.
```
Comment (JosephPJordan) [inline docs/navigate.rst:27]: The **All partition** partition is supported for the VIP table in v2.9 and later.

File: docs/navigate.rst [hunk 4]
```diff
@@ -43,7 +43,7 @@ View Node table
 
 1. Click the Visibility tab, and then click the :guilabel:`Node` table from the Table list.
 
-2. From the :guilabel:`Partition` list, click the partition.
+2. From the :guilabel:`Partition` list, click the partition. **All partition** partition is supported from v2.9 onwards for Node table.
```
Comment (JosephPJordan) [inline docs/navigate.rst:46]: The All partition partition is supported for the Node table in v2.9 and later.

File: docs/navigate.rst [hunk 5]
```diff
@@ -140,8 +140,9 @@ View VIP Dashboard
    
       - It is possible to directly click the **Visibility Dashboard** sub-tab and then select the VIP, instead of a redirect from **Visibility Table**.
       
+5. F5 ACI ServiceCenter supports MAC masquerade mac display under BIG-IP endpoint details on VIP visibility dashboard. MAC address type details( eg. MAC masquerade MAC or Normal MAC ) will be displayed on hover over the information icon in the MAC column of BIG-IP endpoint details table.
```
Comment (JosephPJordan) [inline docs/navigate.rst:143]: (should 'mac' in mac display be capitalized? seems like it.)
... on the VIP visibility dashboard.
MAC address type details (such as MAC masquerade or normal MAC) are displayed when you hover over the Information icon in the MAC column of the BIG-IP endpoint details table.

File: docs/navigate.rst [hunk 6]
```diff
@@ -198,6 +199,13 @@ View Node Dashboard
    c. **APIC Endpoint Details** - The Visibility dashboard displays APIC details for the Node: - 1. MAC 2. EPG 3. Node 4. Interface 5. VLAN Encap 
                                         
    d. **BIG-IP Endpoint Details** - The Visibility dashboard displays BIG-IP details for the Node: - 1. MAC 2. VLAN 3. Interfaces (There is a **View Logs** link beside interfaces to view the interface logs) 4. Self IPs
+   
+6. F5 ACI ServiceCenter supports VXLAN tunnel interfaces along with individual, Port channel, Virtual port channel interfaces on Node table visibility dashboard.
```
Comment (JosephPJordan) [inline docs/navigate.rst:203]: ....along with individual, Port channel, and Virtual port channel interfaces on the Node table visibility dashboard.

File: docs/navigate.rst [hunk 7]
```diff
@@ -198,6 +199,13 @@ View Node Dashboard
...
+   
+   To view details of the VXLAN tunnel interfaces, Click on Node IP in Node information table -> Scroll down to APIC Endpoint details table. All interfaces associated with the node like VPC/PC/Individual/tunnel are displayed in the interface column and On click over the interface additional information related to VPC/PC/Individual/tunnel interface will be displayed in the side panel.
```
Comment (JosephPJordan) [inline docs/navigate.rst:205]: ... tunnel interfaces, click **Node IP** in the Node information table, and then scroll down to the APIC Endpoint details table. 

All interfaces associated with the node, such as VPC, PC, Individual, and Tunnel, are displayed in the interface column. When you click the interface, additional information related to these interfaces displays in the side panel.
(that's an attempt to fix that sentence, I really wasn't sure what '.. and On click over the interface..' meant.)

File: docs/navigate.rst [hunk 8]
```diff
@@ -266,3 +274,9 @@ It displays the **default route domain** for the partition to which the Virtual
 If you encounter this issue, use the 'zoom out' option on your web browser. For example, on Windows, hold the **Ctrl** key, and then click **-** (the dash/minus key).
 
 ------
+
+**Q. VIP MAC masquerade Address might display old MAC on the visibility dashboard**
```
Comment (JosephPJordan) [inline docs/navigate.rst:278]: **Q. VIP MAC masquerade addresses might display an old MAC address on the visibility dashboard**

File: docs/navigate.rst [hunk 9]
```diff
@@ -266,3 +274,9 @@ It displays the **default route domain** for the partition to which the Virtual
...
+
+Due to APIC endpoint learning and retention behavior, it takes 10-15 mins time for updated MAC details to be available on APIC endpoint tracker. Until APIC updates the MAC address, FASC visibility dashboard displays the old MAC address. Refer this link for recommended actions:  https://support.f5.com/csp/article/K44023455
```
Comment (JosephPJordan) [inline docs/navigate.rst:280]: Due to the APIC endpoint learning and retention behavior, it takes 10-15 minutes for the updated MAC details to be available on the APIC endpoint tracker.
Until APIC updates the MAC address, the FASC visibility...

File: docs/pre-req.rst [hunk 10]
```diff
@@ -11,5 +11,20 @@ Pre-requisites for the F5 ACI ServiceCenter:
 - To view the Virtual Server statistics on the Visibility VIP Dashboard, you must install the Telemetry Streaming plugin version 1.17 or higher on the BIG-IP device. Follow the installation steps and the default pull consumer configuration details from https://clouddocs.f5.com/f5-aci-servicecenter/latest/navigate.html#configure-telemetry 
 - To enable and view discovered devices in F5 ACI ServiceCenter, enable the LLDP protocol on both the BIG-IP as well as on the APIC. Make sure management IP of the BIG-IP is present in the LLDP attributes on BIG-IP(Network->Interfaces->Interfaces List-><interface_id>).
 - To configure and manage the Cisco ACI fabric use the APIC GUI/API, example: to add a logical device, to add a end point group etc. The F5 ACI ServiceCenter has read-only privileges to the Cisco APIC.
+- The following BIG-IP Plugin versions that are currently supported by the F5 ACI ServiceCenter:
```
Comment (JosephPJordan) [inline docs/pre-req.rst:14]: The following are the BIG-IP Plugin versions currently supported by the F5 ACI ServiceCenter:

File: docs/release-notes.rst [hunk 11]
```diff
@@ -1,3 +1,277 @@
+Release Notes (Version 2.9)
+===========================
+
+General
+-------
+
+**File system convergence**
+
+APIC Filesystem Glusterfs takes 15 to 20 Minutes to recover when APIC cluster goes unhealthy.
```
Comment (JosephPJordan) [inline docs/release-notes.rst:9]: ..minutes... (no caps)
.. when the APIC cluster goes...

File: docs/release-notes.rst [hunk 12]
```diff
@@ -1,3 +1,277 @@
...
+2. Users may not find previously added BIG-IP entries in the application
+3. New Application installation may fail
+
+**Workaround**: Wait for 20 minutes and try to access the application again. The Glusterfs should recover automatically and the application should be accessible again.
+
+------
+
+**Export db does not work on APIC version 5.0.1**
+
+The user menu on the top-right corner of the F5 ACI ServiceCenter has an option for exporting F5 ACI ServiceCenter database (**Export/Import DB > Export DB**). This option does not work on APIC versions 5.0.1. This is working as expected on prior versions of the APIC and also on versions 5.0(2e) and higher.
```
Comment (JosephPJordan) [inline docs/release-notes.rst:24]: The User menu...   ...for exporting the F5 ACI ...

File: docs/release-notes.rst [hunk 13]
```diff
@@ -1,3 +1,277 @@
...
+**Export db does not work on APIC version 5.0.1**
+
+The user menu on the top-right corner of the F5 ACI ServiceCenter has an option for exporting F5 ACI ServiceCenter database (**Export/Import DB > Export DB**). This option does not work on APIC versions 5.0.1. This is working as expected on prior versions of the APIC and also on versions 5.0(2e) and higher. 
+
+**Workaround:** Backup the F5 ACI ServiceCenter database using the following steps:
+1. Login to the APIC server via SSH.
+
+2. Change directories to F5Networks_F5ACIServiceCenter: **cd /data2/gluster/gv0/F5Networks_F5ACIServiceCenter**.
+
+3. Use scp, winscp or any other preferred tool to copy out the f5.db file from this location.
```
Comment (JosephPJordan) [inline docs/release-notes.rst:31]: ..to copy the f5.db file...

File: docs/release-notes.rst [hunk 14]
```diff
@@ -1,3 +1,277 @@
...
+
+**Zoom In/Out for Device Discovery Topology diagrams may get stuck on Internet Explorer**
+
+When you click the Topology icon in any one of the tabs (Visibility, Network Management, and L4-L7 App Services), the Topology diagrams may get stuck when zooming in and out. The position of the topology diagram can also not be changed once this issue is observed. 
+
+**Workaround:** Use a different browser than IE.
+
+------
+
+**BIG-IP version may not be displayed correctly in FASC when BIG-IP version is upgraded or downgraded.**
```
Comment (JosephPJordan) [inline docs/release-notes.rst:51]: ...when the BIG-IP version...

File: docs/release-notes.rst [hunk 15]
```diff
@@ -1,3 +1,277 @@
...
+
+The BIG-IP version displayed on hover over the BIG-IP IP address may not be displayed correctly when BIG-IP is upgraded or downgraded.
```
Comment (JosephPJordan) [inline docs/release-notes.rst:53]: ..version when hovering over the BIG-IP

File: docs/release-notes.rst [hunk 16]
```diff
@@ -1,3 +1,277 @@
...
+
+**Workaround:** Log-out and re-login the BIG-IP device in the FASC application.
```
Comment (JosephPJordan) [inline docs/release-notes.rst:55]: Log out, and re-login to the BIG-IP device...

File: docs/release-notes.rst [hunk 17]
```diff
@@ -1,3 +1,277 @@
...
+
+------
+
+**For non-admin BIG-IP users, Default Gateway CRUD operations show success even though the corresponding operation on the BIG-IP is not permitted and failed.**
+
+Non-admin BIG-IP users have restricted access to F5 ACI ServiceCenter, however due to an Ansible defect, users might see success messages for Default Gateway operations despite those operations failing on the BIG-IP. 
+
+**Ansible Defect:** https://github.com/F5Networks/f5-ansible/issues/2088
+
+**Workaround:** Use BIG-IP admin account for F5 ACI ServiceCenter Default Gateway operations.
```
Comment (JosephPJordan) [inline docs/release-notes.rst:65]: Use the BIG-IP admin...

File: docs/release-notes.rst [hunk 18]
```diff
@@ -1,3 +1,277 @@
...
+
+1. Switch tabs from the Visibility tab to one of the other tabs and then switch back. 
+
+2. Select a different entry from the Visibility table drop-downs (either Partition or Table) and switch back to the intended combination.
+
+3. Close and re-open the FASC app UI.
+
+------
+
+**Visibility table status icons render few seconds after data for scale configs.**
```
Comment (JosephPJordan) [inline docs/release-notes.rst:105]: (just noticed the period at the end of some of these headings.  Either with or without period is probably fine, but be consistent (ie either remove or add all periods at the end)

File: docs/release-notes.rst [hunk 19]
```diff
@@ -1,3 +1,277 @@
...
+
+**Visibility Dashboard → View Logs** (for both VIP and Node) will display logs in UTC. But for BIG-IP v12, the logs are displayed in local timezone (timezone of the BIG-IP).
+
+**Workaround:** Upgrade BIG-IP to v13 or higher.
+
+------
+
+**Visibility Dashboard filter may display additional logs with interface filter.**
+
+**Visibility Dashboard → BIG-IP Endpoint Details → Interface → View Logs** (for BIG-IP interfaces) displays interface logs. By default it applies the filter of interface name (for example: 1.1 or 1.2); and hence only the logs with interface name in them are displayed. This filtering logic may not work as expected and display additional logs which have interface name as a substring (For example: **1.2** is a substring in the log "Pool /Partition/Application/web_pool member **/Partition/12.14.1.2:8080** monitor status down").
```
Comment (JosephPJordan) [inline docs/release-notes.rst:121]: ...logs which have interface names as a substring...

File: docs/release-notes.rst [hunk 20]
```diff
@@ -1,3 +1,277 @@
...
+------
+
+L4-L7 App Services
+------------------
+
+**Application services declaration not deleted**
+
+If your AS3 declaration contains “optimisticLockKey” mentioned explicitly, the Application Services configuration may not be deleted completely, even after multiple attempts from the application UI. However, the configuration is removed from the BIG-IP device.
+
+**Workaround**: Upload one more AS3 sample declaration to the app and then perform a :guilabel:`Delete all` operation. (Use :guilabel:`View AS3 Declaration` and click :guilabel:`Delete`.)
```
Comment (JosephPJordan) [inline docs/release-notes.rst:134]: ...one or more sample declarations...

File: docs/release-notes.rst [hunk 21]
```diff
@@ -1,3 +1,277 @@
...
+
+-------
+
+**Once all Pool Member cards are removed from template form, adding a new pool member card fails on FASC FAST templates UI**
```
Comment (JosephPJordan) [inline docs/release-notes.rst:138]: ...from the template form,

File: docs/release-notes.rst [hunk 22]
```diff
@@ -1,3 +1,277 @@
...
+
+If user clicks on "-" sign in front on pool members card and tries to add pool member card again by clicking on "+" icon then then pool member card is not populated in template form.
```
Comment (JosephPJordan) [inline docs/release-notes.rst:140]: If a user clicks the "-" sign in front of the pool members card....   ..again by clicking the "+" icon, then the pool member card is not...
(note the "then then" was the cause of the pipeline failure)

File: docs/release-notes.rst [hunk 23]
```diff
@@ -1,3 +1,277 @@
...
+
+**Workaround:** Refresh the basic subtab on L4-L7 tab
```
Comment (JosephPJordan) [inline docs/release-notes.rst:142]: ..on the L4-L7 tab.

File: docs/release-notes.rst [hunk 24]
```diff
@@ -1,3 +1,277 @@
...
+
+**Dynamic hide/show of sub-forms is not supported for FAST templates**
+
+Functionality releated to displaying a sub-form based on some checkbox selection is not supported by FAST forms on F5 ACI ServiceCenter. Currently all the templates are displayed in a completely expanded layout.
+
+**Workaround:** User already has the completely expanded form available on FASC’s FAST UI.
+
+------
+
+**TextBox field displayed for 'Notice: Beta Test' field in 'bigip-fast-templates/microsoft adfs' and 'bigip-fast-templates/microsoft exchange' FAST templates.**
```
Comment (JosephPJordan) [inline docs/release-notes.rst:154]: A TextBox  field is displayed for ... ...FAST template, but does not take data
(or something like that, it's odd without something about the issue in the heading)

File: docs/release-notes.rst [hunk 25]
```diff
@@ -1,3 +1,277 @@
...
+
+**Workaround:** Click the **Submit** button again, and the mapping will be reset properly without any errors. 
+
+**AS3 Defect:** https://github.com/F5Networks/f5-appsvcs-extension/issues/185
+ 
+------
+
+**AS3 applications can either have static nodes or dynamic nodes but not both**
+
+AS3 applications will support either static nodes or dynamic nodes (using the **Manage Endpoint Mappings** button) but not both.
```
Comment (JosephPJordan) [inline docs/release-notes.rst:186]: (does this need a "workaround: none" entry?  )

File: docs/release-notes.rst [hunk 26]
```diff
@@ -1,3 +1,277 @@
...
+
+------
+
+**Dynamic EP discovery does not work if a duplicate IP already exists on a different partition.**
+
+If an APIC Tenant|App|EPG mapped to a BIG-IP pool has an endpoint with an IP address which already exists on the BIG-IP but in a different partition, then the APIC endpoint will not get added to BIG-IP pool. Also any successive configurations and endpoints also will not be discovered/deleted from this BIG-IP pool. 
+
+Workaround: Remove the duplicate IPs from the endpoint list on APIC and retry a manual sync of Endpoints from L4-L7 App Services --> Application Inventory --> Sync EPs icon. 
+
+Note: Similar issues might be seen with other erronous configurations such as unsupported IPv4 formats like 1.2.3.4/24 instead of 1.2.3.4
```
Comment (JosephPJordan) [inline docs/release-notes.rst:246]: erroneous

**Review by JosephPJordan — CHANGES_REQUESTED** [review]
some suggested edits and comments

**Comment by JosephPJordan** [comment]
@vishwajeetgslab, note the pipeline failure will be fixed when you implement my edits, it failed due to a repeated word in the Grammar check

File: docs/install.rst [hunk 27]
```diff
@@ -122,11 +122,15 @@ Add a new BIG-IP device (Device Login)
 
 4. Log in to the BIG-IP device. The device hostname, redundancy state, and config sync state are displayed at the top of the page, along with three tabs: Visibility, L2-L3 Stitching, and L4-L7 Configuration.
 
+5. F5 ACI ServiceCenter v2.9 onwards supports LDAP authentication for BIG-IP login.
```
Comment (manuwelakanade-GSLAB) [inline docs/install.rst:125]: F5 ACI ServiceCenter (v2.9+) supports LDAP authentication for BIG-IP login. Admin LDAP users will be able to use all the features of the application.

File: docs/navigate.rst [hunk 28]
```diff
@@ -10,11 +10,11 @@ View VLAN table
 
    In the :guilabel:`Table` list, the VLAN table is selected by default.
 
-2. From the :guilabel:`Partition` list, select the partition you’re interested in.
+2. From the :guilabel:`Partition` list, select the partition you’re interested in. The **All partition** partition is supported for the VLAN table in v2.9 and later.
```
Comment (manuwelakanade-GSLAB) [inline docs/navigate.rst:13]: The **All partitions** option is supported for the VLAN table in v2.9 and later.

File: docs/navigate.rst [hunk 29]
```diff
@@ -24,7 +24,7 @@ View VIP table
 
 1. Click the Visibility tab, and then from the :guilabel:`Table` list, click the :guilabel:`VIP` table.
 
-2. From the :guilabel:`Partition` list, click the partition you're interested in.
+2. From the :guilabel:`Partition` list, click the partition you're interested in. The **All partition** partition is supported for the VIP table in v2.9 and later.
```
Comment (manuwelakanade-GSLAB) [inline docs/navigate.rst:27]: The **All partitions** option is supported for the VIP table in v2.9 and later.

File: docs/navigate.rst [hunk 30]
```diff
@@ -43,7 +43,7 @@ View Node table
 
 1. Click the Visibility tab, and then click the :guilabel:`Node` table from the Table list.
 
-2. From the :guilabel:`Partition` list, click the partition.
+2. From the :guilabel:`Partition` list, click the partition. **All partition** partition is supported for the Node table in v2.9 and later.
```
Comment (manuwelakanade-GSLAB) [inline docs/navigate.rst:46]: The **All partitions** option is supported for the Node table in v2.9 and later.

File: docs/pre-req.rst [hunk 31]
```diff
@@ -11,5 +11,20 @@ Pre-requisites for the F5 ACI ServiceCenter:
 - To view the Virtual Server statistics on the Visibility VIP Dashboard, you must install the Telemetry Streaming plugin version 1.17 or higher on the BIG-IP device. Follow the installation steps and the default pull consumer configuration details from https://clouddocs.f5.com/f5-aci-servicecenter/latest/navigate.html#configure-telemetry 
 - To enable and view discovered devices in F5 ACI ServiceCenter, enable the LLDP protocol on both the BIG-IP as well as on the APIC. Make sure management IP of the BIG-IP is present in the LLDP attributes on BIG-IP(Network->Interfaces->Interfaces List-><interface_id>).
 - To configure and manage the Cisco ACI fabric use the APIC GUI/API, example: to add a logical device, to add a end point group etc. The F5 ACI ServiceCenter has read-only privileges to the Cisco APIC.
+- The following are the BIG-IP Plugin versions currently supported by the F5 ACI ServiceCenter:
```
Comment (manuwelakanade-GSLAB) [inline docs/pre-req.rst:14]: BIG-IP Plugin versions currently supported by the F5 ACI ServiceCenter:

**Review by manuwelakanade-GSLAB — CHANGES_REQUESTED** [review]
@vishwajeetgslab : Requesting few changes.

**Comment by vishwajeetgslab** [comment]
Hi @manuwelakanade-GSLAB @JosephPJordan Thanks for review. i have made the changes as per review comments.

@JosephPJordan Can you please merge the request. Thanks

**Review by JosephPJordan — APPROVED** [review]

**Pull request merged**